test/testsuites/cscpackagetests.go

```diff
@@ -0,0 +1,123 @@
+// created.
+func cscWithNonExistingPackage(t *testing.T, f *test.Framework, ctx *test.TestCtx) error {
+	// Get test namespace
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("Could not get namespace: %v", err)
+	}
+
+	// nonExistingPackageCSCName is the name of the catalogsourceconfig that points
+	// to a non-existing Package
+	nonExistingPackageCSCName string = "non-existing-package-operators"
+
+	// packageName is the non-existing package
+	packageName string = "non-existing-package"
+
+	// Create a new catalogsourceconfig with a non-existing Package
+	nonExistingPackageCSC := &operator.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.CatalogSourceConfigKind,
+		}, ObjectMeta: metav1.ObjectMeta{
+			Name:      nonExistingPackageCSCName,
+			Namespace: namespace,
+		},
+		Spec: operator.CatalogSourceConfigSpec{
+			TargetNamespace: namespace,
+			Packages:        packageName,
+		}}
+	err = helpers.createRuntimeObject(f, ctx, nonExistingPackageCSC)
+	if err != nil {
+		return err
+	}
+
+	// Check that we created the catalogsourceconfig with a non-existing package
+	resultCatalogSourceConfig := &operator.CatalogSourceConfig{}
+	err = helpers.WaitForResult(t, f, resultCatalogSourceConfig, namespace, nonExistingPackageCSCName)
+	if err != nil {
+		return err
+	}
+
+	// Check if the catalogsourceconfig phase and message are the expected values
+	expectedPhase := "Configuring"
+	expectedMessage := fmt.Sprintf("Failed to find package: %v", packageName)
+	// Check that the catalogsourceconfig with an non-existing package eventually reaches the
+	// configuring phase with the expected message
+	err = wait.Poll(retryInterval, timeout, func() (bool, error) {
+		err = WaitForResult(t, f, resultCatalogSourceConfig, namespace, nonExistingPackageCSCName)
+		if err != nil {
+			return false, err
+		}
+		if resultCatalogSourceConfig.Status.CurrentPhase.Name == expectedPhase &&
+			resultCatalogSourceConfig.Status.CurrentPhase.Message == expectedMessage {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return fmt.Errorf("CatalogSourceConfig never reached expected phase/message, expected %v/%v", expectedPhase, expectedMessage)
+	}
+
+	catalogSourceName := nonExistingPackageCSCName
+	serviceName := nonExistingPackageCSCName
+	deploymentName := nonExistingPackageCSCName
+
+	// Check catalog source was not created.
+	resultCatalogSource := &olm.CatalogSource{}
+	err = helpers.WaitForResult(t, f, resultCatalogSource, namespace, catalogSourceName)
+	if err == nil {
+		return fmt.Errorf("CatalogSourceConfig with non-existing package created a CatalogSource")
+	}
+
+	// Check service was not created.
+	resultService := &corev1.Service{}
+	err = helpers.WaitForResult(t, f, resultService, namespace, serviceName)
+	if err == nil {
+		return fmt.Errorf("CatalogSourceConfig with non-existing package created a Service")
+	}
+
+	// Check deployment was not created.
```

Comment:
The three checks will result in this test taking 15 minutes to complete. That is not acceptable.

Reply:
makes sense - so just launching them in parallel is ok?

Reply:
I vaguely remember having a discussion with @awgreene about parametrizing this.

Reply:
ok will follow up

Reply:
>I vaguely remember having a discussion with @awgreene about parametrizing this.

That conversation was from earlier in this thread: https://github.com/operator-framework/operator-marketplace/pull/151#discussion_r273581053